streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -61,7 +61,7 @@ public StandbyTask(TaskId id,
         // initialize the topology with its own context
         this.processorContext = new StandbyContextImpl(id, applicationId, config, stateMgr, metrics);
 
-        initializeStateStores();
+        registerAndinitializeStateStores();
```

Comment:
I think we can work around it: `ProcessorStateManager.checkpointedOffsets()` relies on variable `checkpointedOffsets` and `restoreCallbacks`; `checkpointedOffsets` is initialized upon the creation of the manager object, and `restoreCallbacks` can be just replace with `stores`, as we only need its topic names, which is available after registration.